cpp/src/parquet/column_reader.cc

```diff
@@ -940,7 +940,7 @@ int64_t TypedColumnReaderImpl<DType>::ReadBatchWithDictionary(
     int64_t* indices_read, const T** dict, int32_t* dict_len) {
   bool has_dict_output = dict != nullptr && dict_len != nullptr;
   // Similar logic as ReadValues to get pages.
-  if (!HasNext()) {
+  if (batch_size == 0 || !HasNext()) {
```

Comment:
this change breaks use-case in my company where we use this API with batch_size = 0 explicitly, just to obtain dictionary alone. we use it for our loading .parquet files into properitary in-memory column store flow.

it is ok from our perspective to change it as we can just add dedicated API to obtain dictionary in our fork of the code.

i am just bringing this up in case there are other devs impacted by the public API change.

https://www.hyrumslaw.com/

Reply:
We should probably support this. I think the more important question is should we throw an exception for bad pages or just skip over the phantom values. Parquet-mr seems to do the latter but does other validation on the reported value counts. It would also make the batch size 0 case easier.

Reply:
Changed check to allow batch size 0 when reading dict.